### PR TITLE
[CPT-2187] Allow qb controls customization from consumers

### PR DESCRIPTION
**Pull request by angelinastavniiciuc**
[CPT-2187](https://toptal-core.atlassian.net/browse/CPT-2187)

### Description

In this PR we added new property in order to provide desired text/functionality to Run Query button. Considering that `totalCount` and `totalCountLoading` props are something custom that should not be defined on the level of library we removed these props from Query Builder.
Also we removed deprecation of `hideControls` property that was introduced before. There was a misalignment and at the moment we are not planning to move controls to consumers because Run Query and Clear Query it is something common. Instead not showing these controls it is an edge case which is handled by `hideControls`.
 
### How to test

- [Temploy](https://picasso.toptal.net/CPT-2187-allow-qb-controls-customization-from-consumers)
- Check Custom Run Query Button story
### Screenshots

| After.                                  |
 | --------------------------------------- |
 | Screenshot from 2024-04-25 10-53-34 |

### Development checks

- [x] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [x] Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [x] Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)

**Breaking change**

- n/a codemod is created and showcased in the changeset
- [x] test alpha package of Picasso in StaffPortal

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[CPT-2187]: https://toptal-core.atlassian.net/browse/CPT-2187?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by angelinastavniiciuc** [comment]
@toptal-anvil ping reviewers

**Review by TomasSlama — APPROVED** [review]

**Comment by angelinastavniiciuc** [comment]
@toptal-bot run package:alpha-release

File: packages/picasso-query-builder/src/QueryBuilder/QueryBuilder.tsx [hunk 1]
```diff
@@ -59,23 +59,18 @@ type Props = {
...
-  /**
-   * @deprecated [CPT-2188] Controls will be defined at the consumer level
-   * Defines the possibility to display, or not, any of the controls. For example "Clear query" or "Run query" control.
-   */
+  /** Defines the possibility to display, or not, any of the controls. For example "Clear query" or "Run query" control. */
   hideControls?: boolean
   /** Defines the possibility to enable, or not, drag-and-drop functionality. This possibility applies to rules and groups to rearrange it within QB. */
   enableDragAndDrop?: boolean
+  /** Defines custom Run Query button children that allows to change button text or display custom logic. */
+  runQueryChildren?: ReactNode
```
Comment (toptalwadiibasmi) [inline packages/picasso-query-builder/src/QueryBuilder/QueryBuilder.tsx:67]: Nit: runQueryContent as in "the content of run query" seems to be a more expressive option than the "children of run query" wdyt?
```suggestion
  runQueryContent?: ReactNode
```

**Review by toptalwadiibasmi — APPROVED** [review]
Looks good! 

 have one comment regarding the name of the property for better expression of its usage.

**Review by dmaklygin — APPROVED** [review]
Good job!

**Comment by angelinastavniiciuc** [comment]
@toptal-bot run package:alpha-release

**Comment by angelinastavniiciuc** [comment]
@toptal-bot run package:alpha-release

**Review by rasitozcan — APPROVED** [review]

File: packages/picasso-query-builder/src/QueryBuilder/QueryBuilder.tsx [hunk 2]
```diff
@@ -59,23 +59,18 @@ type Props = {
...
-  /**
-   * @deprecated [CPT-2188] Controls will be defined at the consumer level
-   * Defines the possibility to display, or not, any of the controls. For example "Clear query" or "Run query" control.
-   */
+  /** Defines the possibility to display, or not, any of the controls. For example "Clear query" or "Run query" control. */
   hideControls?: boolean
   /** Defines the possibility to enable, or not, drag-and-drop functionality. This possibility applies to rules and groups to rearrange it within QB. */
   enableDragAndDrop?: boolean
+  /** Defines custom Run Query button content that allows to change button text or display custom logic. */
+  runQueryButtonContent?: ReactNode
```
[the remark this answers is not in the document]
Comment (angelinastavniiciuc) [reply packages/picasso-query-builder/src/QueryBuilder/QueryBuilder.tsx:67]: thanks! I suppose that it is okay considering previous reviews
